wallet-web/components/bond/BondNodeForm.tsx

```diff
@@ -1,380 +1,480 @@
-            const bytes = bs58.decode(key);
-            // of length 32
-            return bytes.length === 32
-        } catch {
-            return false
-        }
+    // just AND everything together
+    const reducer = (acc, current) => acc && current
+    return Object.entries(newValidity)
+      .map((entry) => entry[1])
+      .reduce(reducer, true)
+  }
+
+  const validateAmount = (rawValue: string): boolean => {
+    // tests basic coin value requirements, like no more than 6 decimal places, value lower than total supply, etc
+    if (!basicRawCoinValueValidation(rawValue)) {
+      return false
     }
 
-    const validateHost = (host: string): boolean => {
-        // I don't think that proper checks are in scope of the change here
-        // what would need to be checked is whether one of the following is true:
-        // - host is an ipv4 address
-        // - host is an ipv6 address
-        // - host is a valid hostname
-
-        // so at least perform the dumbest possible checks
-        // ipv4 needs 4 dot-separated octets
-        // ipv6 can have multiple possible representations, but it needs to contain at least two colons
-        // a hostname (in this case) needs to have a top level domain present
-
-        const dot_occurrences = host.split('.').length - 1
-        const colon_occurrences = host.split(':').length - 1
-
-        if (dot_occurrences == 3) {
-            // possible ipv4
-            // make sure it has no ports attached!
-            return colon_occurrences == 0
-        } else if (colon_occurrences >= 2) {
-            // possible ipv6
-            return true
-        } else if (dot_occurrences >= 1) {
-            // possible hostname
-            // make sure it has no ports attached!
-            return colon_occurrences == 0
-        }
-        return false
+    // this conversion seems really iffy but I'm not sure how to better approach it
+    let nativeValueString = printableBalanceToNative(rawValue)
+    let nativeValue = parseInt(nativeValueString)
+    if (props.type == NodeType.Mixnode) {
+      return nativeValue >= parseInt(props.minimumMixnodeBond.amount)
+    } else {
+      return nativeValue >= parseInt(props.minimumGatewayBond.amount)
     }
-
-    const validateVersion = (version: string): boolean => {
-        // check if its a valid semver
-        return semver.valid(version) && semver.minor(version) >= 11
-
+  }
+
+  const validateKey = (key: string): boolean => {
+    // it must be a valid base58 key
+    try {
+      const bytes = bs58.decode(key)
+      // of length 32
+      return bytes.length === 32
+    } catch {
+      return false
     }
-
-    const validateLocation = (location: string): boolean => {
-        // right now only perform the stupid check of whether the user copy-pasted the tooltip... (with or without brackets)
-        return !location.trim().includes("physical location of your node")
+  }
+
+  const validateHost = (host: string): boolean => {
```

Comment:
This is not an error on your end, but omission in what I've done - could you check for white spaces at beginning (and end) of the host string? I've noticed people were bonding with for example `<SPACE-HERE>1.1.1.1` value. Or maybe just trim it for them.

Reply:
I've updated this. Hopefully that's what you're after.